espnet2/bin/s2t_inference.py

```diff
@@ -269,40 +274,59 @@ def __init__(
             length_bonus=penalty,
             scorefilter=1.0,
         )
-        beam_search = BeamSearch(
-            beam_size=beam_size,
-            weights=weights,
-            scorers=scorers,
-            sos=s2t_model.sos,
-            eos=s2t_model.eos,
-            vocab_size=len(token_list),
-            token_list=token_list,
-            pre_beam_score_key=None if ctc_weight == 1.0 else "full",
-            normalize_length=normalize_length,
-        )
+        if partial_ar:
+            beam_search_class = PartiallyARInference
+            beam_search = beam_search_class(
```

Comment:
We can remove the first line and directly use `PartiallyARInference` in the second line.